File: 2020/go/01.go
package main

import (
	"io/ioutil"
	"regexp"
	"strconv"
)

func getInput(input string) []string {
	regex := regexp.MustCompile("(\\d+)")
	return regex.FindAllString(input, -1)
}

func answer1(inputStr string) int {
	input := getInput(inputStr)
	for i1, v1 := range input {
		for i2, v2 := range input {
			if i1 != i2 {
				val1, _ := strconv.Atoi(v1)
				val2, _ := strconv.Atoi(v2)
				if val1+val2 == 2020 {
					return val1 * val2
				}
			}
		}
	}
	return -1
}

func answer2(inputStr string) int {
	input := getInput(inputStr)
	for i1, v1 := range input {
		for i2, v2 := range input {
			for i3, v3 := range input {
				if i1 != i2 && i1 != i3 {
					val1, _ := strconv.Atoi(v1)
					val2, _ := strconv.Atoi(v2)
					val3, _ := strconv.Atoi(v3)
					if val1+val2+val3 == 2020 {
						return val1 * val2 * val3
					}
				}
			}
		}
	}
	return -1
}
func main() {
	input, _ := ioutil.ReadFile("01input.txt")
	println(answer1(string(input)))
	println(answer2(string(input)))
}

File: 2020/lua/01.lua
local utils = require"utils"

local INPUT = utils.readFile("01.input")

function answer1(input)
    local inputTable = utils.split(input, utils.lineDelimiter)

    for i = 1, #inputTable-2 do
        local v1 = inputTable[i]
        for j = i+1, #inputTable-1 do
            local v2 = inputTable[j]
            if v1 + v2 == 2020 then
                return v1 * v2
            end
        end
    end
end

function answer2(input)
    local inputTable = utils.split(input, utils.lineDelimiter)
    for i = 1, #inputTable-2 do
        for j = i+1, #inputTable-1 do
            for k = j+1, #inputTable do
                local v1 = inputTable[i]
                local v2 = inputTable[j]
                local v3 = inputTable[k]
                if v1 + v2 + v3 == 2020 then
                    return v1 * v2 * v3
                end
            end
        end
    end
end

print("The answer 1 is", answer1(INPUT))
print("The answer 2 is", answer2(INPUT))